apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SessionPoolManager.cs

```diff
@@ -313,7 +323,8 @@ private static async Task<SpannerClient> CreateClientAsync(SpannerClientCreation
             return new SpannerClientBuilder
             {
                 CallInvoker = callInvoker,
-                Settings = spannerSettings
+                Settings = spannerSettings,
+                EmulatorDetection = emulatorDetection
```

Comment:
Setting a call invoker and emulator detection would be an odd thing to do. (I'm slightly surprised we don't prohibit it.)
If we do the work in SpannerClientCreationOptions (which may mean some code duplication, unfortunately - let's see what we can do) the endpoint and credentials will already be set correctly, so we shouldn't need to make any changes here.

Reply:
Oh I see, so we shouldn't rely on the automated emulator detection for the data client. We should only do that for the admin clients. So we'll manually read the env var and set the endpoint and credentials. Is the worry that it would confuse users because they would set an endpoint and credentials but we'd be using a different parameters for the actual connection?

Reply:
It's not a matter of "not relying" on the automated emulation code - it's not that we don't trust it. It's a matter of making it fit in well with the rest of the infrastructure.
I'd like to investigate how to do this without code duplication though - for example, it may be possible to add a *public* member to `SpannerClientBuilder` to say "please let me know how you would be configured for emulation" (effectively like the current partial method does) and then use the endpoint and credentials from that.

I think our aim should be for the behavior to be consistent between "using SpannerClientBuilder directly" and "using SpannerConnectionStringBuilder directly" in terms of the connection.

Reply:
I've made the minimal amount of changes to get this working. But I'm still not updating the endpoint and credentials in the connection string to reflect what is actually being connected to. I think this is where your comment regarding updating the `SpannerClientBuilder` comes in.

The other issue that I want to solve is that if the connection string has EmulatorDetection then users shouldn't be asked to specify credentials. I'll also work on trying to bypass this check.